src/ufuncs.c

```diff
@@ -530,6 +530,18 @@ static PyUFuncGenericFunction Y_i_funcs[1] = {&Y_i_func};
 /* Define the geom, geom -> double functions (YY_d) */
 static void *distance_data[1] = {GEOSDistance_r};
 static void *hausdorff_distance_data[1] = {GEOSHausdorffDistance_r};
+#if GEOS_SINCE_370
+  static int GEOSFrechetDistanceWrapped_r(void *context, void *a,  void *b, double *c) {
+      /* Handle empty geometries (they give segfaults) */
+      if (GEOSisEmpty_r(context, a) | GEOSisEmpty_r(context, b)) {
+          *c = NPY_NAN;
```

Comment:
Suggestion: shift the logic around to keep the bare return the happy path:

```
if (GEOSisEmpty_r(context, a) | GEOSisEmpty_r(context, b)) {
    *c = NPY_NAN;
    return 1
}

return GEOSFrechetDistance_r(context, a, b, c);

Reply:
Done.